iterx.go

```diff
@@ -132,10 +143,17 @@ func (iter *Iterx) scanAll(dest interface{}) bool {
 	base := reflectx.Deref(slice.Elem())
 	scannable := isScannable(base)
 
-	// if it's a base type make sure it only has 1 column;  if not return an error
-	if scannable && len(iter.Columns()) > 1 {
-		iter.err = fmt.Errorf("non-struct dest type %s with >1 columns (%d)", base.Kind(), len(iter.Columns()))
-		return false
+	if structScan {
```

Comment:
If `structScan` is true there is no need to check if it's scannable right? Is there a chance for an early return when the user wants to scan into a struct?